### PR TITLE
Enable user registration

### PR DESCRIPTION
**Pull request by buehner**
Enable user registration by introducing:

* a `RegistrationToken`
* a web interface `/user/register.action` that accepts an email address and a password
  * An email with an activation link will be send to the passed email address
* a web interface `/user/activate.action` that accepts a token value (the link will be send in the email from above)
  * The user will be activated and the default user role `ROLE_USER` is being assigned to the user. The user can login afterwards.

Also added some missing tests

### CODE REVIEW

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	 */
+	public RegistrationToken findByUser(User user) {
+
+		SimpleExpression eqUser = Restrictions.eq("user", user);
+
+		RegistrationToken registrationToken =
+				dao.findByUniqueCriteria(eqUser);
+
+		return registrationToken;
+	}
```
Comment (KaiVolland) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java]: Why not using the `findByUser()` presented by the abstract service class?

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	 */
+	public RegistrationToken findByTokenValue(String token) {
+
+		Criterion criteria = Restrictions.eq("token", token);
+
+		RegistrationToken registrationToken =
+				dao.findByUniqueCriteria(criteria);
+
+		return registrationToken;
+	}
```
Comment (KaiVolland) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java]: See above.

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+ * @author Nils Bühner
+ *
+ */
+@Service("registrationTokenService")
+public class RegistrationTokenService extends AbstractUserTokenService<RegistrationToken> {
+
+	/**
+	 * The relative URL for the SHOGun2 registration activation interface.
+	 */
+	private static final String REGISTER_ACTIVATION_URL = "/user/activate.action";
```
Comment (KaiVolland) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java]: Please make configurable!

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java (hunk 1 above)
Comment (buehner) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java]: Ooops, just forgot to remove them from here :+1:

File: src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java (hunk 2 above)
Comment (buehner) [inline src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/RegistrationTokenService.java]: Ooops, just forgot to remove them from here :+1:

File: src/shogun2-webapp-archetype/src/main/resources/archetype-resources/src/main/resources/META-INF/__artifactId__.properties [hunk 4]
```diff
@@ -23,7 +23,7 @@ mail.defaultSender=noreply@${artifactId}.de
 ${symbol_pound} The mail template being used as confirmation mail after registration
 mail.registrationMailTemplateSubject=[${artifactId}] Activate your account
 mail.registrationMailTemplateText=${symbol_escape}
-Dear %s,${symbol_escape}n${symbol_escape}
+Dear user,${symbol_escape}n${symbol_escape}
 ${symbol_escape}n${symbol_escape}
 Welcome to ${artifactId}!${symbol_escape}n${symbol_escape}
 ${symbol_escape}n${symbol_escape}
```
Comment (KaiVolland) [inline src/shogun2-webapp-archetype/src/main/resources/archetype-resources/src/main/resources/META-INF/__artifactId__.properties]: Set time range from 48h to 60 minutes (or delete it).

**Comment by KaiVolland** [comment]
:+1:  plz merge @dnlkoch @weskamm like it also

**Comment by buehner** [comment]
I added three commits, that address the mentioned issues!

On top of that, the expiration time for a registration token is configurable now.

**Comment by marcjansen** [comment]
:+1: from me, please merge at will once the checks are green.

**Pull request merged**